### DISCUSSION TITLE
Code Clubs

### DISCUSSION THREAD

**Discussion by metazool** [ukceh-rse/community#15] [General]
Code clubs, or code clinics are a regular, in-person forum for everyone working on research code to share knowledge and solve problems.

The model I'm proposing for running them is based on what happens at the [British Geological Survey](https://github.com/BritishGeologicalSurvey/) - there are some [notes on that here](https://github.com/ukceh-rse/community/blob/main/clinics.md)

* Fortnightly meetup at a regular time and place
* Facilitated by, but not always run by Research Software Engineers
* Varying format - could be based on an exercise, a topic, or "bring your own problem"
* Emphasis on hands-on demonstration over presentation

In the Edinburgh site there's a Data Users Group which is mostly online-based, and used to do more in-person activity, it makes sense to start a Code Club meetup as part of that existing community. What's like this at other sites?

**Comment by metazool** [comment]
See also #16 - suggestion for a few "Pipelines - how to get started" in different languages and frameworks sessions

**Comment by samharrison7** [comment]
At Lancaster we have Stats Lunch that has been running for years and I think it is well attended (though I think I've only managed to attend 2 of them in the 7 years I've been here!). As the name implies, it's mainly stats stuff and has a heavy bias towards R.

There's also a CEEDS Coding and Stats group that's been going in various guises since about 2021. I think the focus these days is again on R. I few of us tried to get a Python version going back in 2021 (https://github.com/ceeds-coding-club/use-python), but we got hardly any attendees and so it fell by the wayside.

I like the idea of a Code Clinic. The main reason I don't attend the Stats Lunch or CEEDS ones is that I only very rarely use R, so it would be good to make them language-agnostic.

**Reply by metazool** [reply]
I'd forgotten about this link to the CEEDS club, @samharrison7 but it came up again today during a regular gathering of the "Science Skills Squad" - 

You can tell from the state of the R and python repositories where the momentum is! https://github.com/ceeds-coding-club/ - the python focused looks like it is yourself and @zhangce8880

Do you know if the regular in-person meetings in Lancaster are still happening? It definitely makes more sense for our RSE team members based there to connect with them rather than initiate something in parallel.

https://github.com/ukceh-rse/community/discussions/26 - notes and followup links from our Edinburgh session next week. We just wheeled a big screen into a sofa space. This was an R project and neither @jmarshrossney or i have a particular R strength but could still provide a technical judgement sounding board and some pointers towards other people. So, language-agnostic, bring us Fortran, containerisation queries, modelling questions, anything at all and we will try to help figure it out :)

**Reply by samharrison7** [reply]
Yeah, the Python group rapidly lost momentum... We didn't get many people attending. Couldn't really figure out why - it's difficult to believe there are fewer people into Python than R!

The in-person meetings are still happening for R, but not Python. Supreeta Vijayakumar from LEC coordinates them now and she sends emails to the CEEDS mailing list with details. I've never been along to one though.

Personally I would like language-agnostic sessions - I feel there is a lot to be learnt from someone demo-ing something in a language you're not familiar with, particularly to make you think about whether the language you're writing in is appropriate for the task at hand (i.e. "oh that looks _soo_ much easier in `<insert-language-name-here>`, why aren't I doing that?!")

**Comment by metazool** [comment]
suggestion in the Q&A about advances and norms in python packaging as a good topic for these sessions (nice for sites where there is already a lot of research coding in python, also for engaging EDS dev teams)

**Comment by susheelbhanu** [comment]
This is a great idea, however not everybody works on site so having a dedicated timeslot would help. Additionally, it may be more inclusive if an online session was run, every once in a while.

**Comment by metazool** [comment]
We should learn about, and from, the past experience of the Stats Network ...

**Comment by metazool** [comment]
Another topic here is python environment management, different tools, helping people find good paths
